hail/python/hailtop/batch/contrib/regenie/regenie-batch.py

```diff
@@ -0,0 +1,293 @@
+    parser.add_argument('--spa', required=False, action='store_true')
+    parser.add_argument('--debug', required=False, action='store_true')
+    parser.add_argument('--verbose', required=False, action='store_true')
+    parser.add_argument('--lowmem', required=False, action='store_true')
+
+    parser.add_argument('--lowmem-prefix', required=False)
+    parser.add_argument('--threads', required=False, default=1)
+
+    if not hexists(path_or_str):
+        print(f"Couldn't find a file named {path_or_str}, assuming this is an argument string")
+        r = parser.parse_args(shlex.split(path_or_str))
+
+    else:
+        print(f"Found {path_or_str}, reading")
+
+        with hopen(path_or_str, "r") as f:
+            t = shlex.split(f.read())
+            r = parser.parse_args(t)
+
+    ninputs = sum(i is not None for i in [r.bed, r.bgen, r.pgen])
+    if ninputs == 0:
+        raise Exception("An input file is required. Please specify: --bed or --bgen or --pgen")
+
+    if ninputs > 1:
+        raise Exception(f"""More than one input file specified.
+                            Check that only one of --bed or --bgen or --pgen are specified for --step {r.step}
+                        """)
+
+    if r.step == 2:
+        if r.pred:
+            print("--pred provided for --step 2, but Batch will constrain the --pred output prefix to that of --step 1.")
+
+        if r.extract or r.exclude:
+            raise Exception("--extract and --exclude only work with --step 1")
+
+    if r.lowmem and not r.lowmem_prefix:
+        raise Exception("When --lowmem provided, --lowmem-prefix required")
+
+    return r
+
+
+# TODO: check --phenoCol, --phenoColList, --remove
+def get_phenos(step_args: argparse.Namespace):
+    with hopen(step_args.phenoFile, "r") as f:
+        return f.readline().strip().split(" ")[2:]
+
+
+def get_input(batch, step_args: argparse.Namespace):
+    add = {}
+    for name, val in vars(step_args).items():
+        if name in from_underscore:
+            name = from_underscore[name]
+
+        if name not in input_file_args or val is None:
+            continue
+
+        if name == "bed":
+            prefix = step_args.bed
+            inf = {}
+            inf["bed"] = f"{prefix}.bed"
+            inf["bim"] = f'{prefix}.bim'
+            inf["fam"] = f'{prefix}.fam'
+            add[name] = batch.read_input_group(**inf)
+            continue
+
+        if name == "pgen":
+            prefix = step_args.pgen
+            inf = {}
+            inf["pgen"] = f'{prefix}.pgen'
+            inf["pvar"] = f'{prefix}.pvar'
+            inf["psam"] = f'{prefix}.psam'
+            add[name] = batch.read_input_group(**inf)
+            continue
+
+        add[name] = batch.read_input(val)
+
+    return add
+
+
+def prepare_jobs(batch, args: BatchArgs, step1_args: argparse.Namespace,
+                 step2_args: argparse.Namespace):
+    j1 = batch.new_job(name='run-regenie')
+    j1.image('akotlar/regenie:latest')
```

Comment:
Hmm. We probably want this to be a script option and we'll want to build an image in `build.yaml` that we can use in CI tests to test `regenie-batch.py`. I also understand that we're still ironing out details here. We'll want the Hail deploy process to push out a copy of this image versioned by the hail pip version, e.g. `hailgenetics/regenie:0.2.53`.

I'm really inclined to have automated tests. How far are we from being able to run this on a tiny (like 10 variants and 10 samples, say) and verify the results? It sounds like you've been running tests already, are those hard to get integrated into an automated test as a part of this PR? We don't have to verify outputs for the first PR, we can just verify REGENIE runs without error.

Reply:
As of this PR the "demo" version works and outputs the correct file. I'll push a test today (as well as fixes for most of your comments, thanks for those)

Reply:
The image should just go to hail genetics; I am not sure I have permissions to push there, will check when I get home. I could not push to hail-vdc or broad-ctsa projects in my first attempt, and didn't bother working out the permissions issues as I had something that worked better for the initial pr